### ISSUE TITLE
Multiprocess decompression?

### ISSUE THREAD

**Issue by randocolab** [animalize/pyzstd#15]
Hi, I'm using `pyzstd` in combination with `tarfile` to read & parse a large amount of small XML files.
I tried to parallelize using  `Multiprocessing`, that's when I ran into a issue:
`multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7feb229d3700>'. Reason: 'PicklingError("Can't pickle <class '_zstd.ZstdError'>: import of module '_zstd' failed")'`
I vaguely understand that multiprocessing is trying to pickle pyzstd classes which it fails at..
I tried moving the `with pyzstd.open(file_path) as archive:` to the `progresser` function is slower than the single threaded version (150 iterations/s).

Could you give me some insight/point me in the right direction on how to tackle this problem?

This is my setup
```python
import multiprocessing as mp
import xml.etree.ElementTree as ET
import pyzstd
from tqdm import tqdm

with pyzstd.open(file_path) as archive:
    tar_file = tarfile.open(fileobj=archive)


    def progresser(XML):
        parsed = ET.parse(tar_file.extractfile(XML))
        # do things with parsed result ...


    with mp.Pool() as pool:
        results = tqdm(
            pool.imap_unordered(progresser, files, chunksize=12),
            total=len(files),
        )

        for result in results:
            pass

# single threaded version: 1510 Iterations/s
with pyzstd.open(file_path) as archive:
    tar_file = tarfile.open(fileobj=archive)
    for file in tqdm(tar_file):
        ET.parse(tar_file.extractfile(file))

```

If this example is hard to follow/read I can give another one.

**Comment by animalize** [comment]
Thanks for your report, fixed in e8cd9c662604509227dbdf2ed50d9530f5dd0f5f.

To support C/cffi implementations, pyzstd uses two-level directory structure, which leads to `_zstd.ZstdError` can't be located.

**Comment by animalize** [comment]
0.15.3 released.
